code/game/objects/items/weapons/powerfist.dm

```diff
@@ -0,0 +1,100 @@
+/obj/item/weapon/melee/powerfist
+	name = "power-fist"
+	desc = "A metal gauntlet with a piston-powered ram ontop for that extra 'ompfh' in your punch."
+	icon_state = "powerfist"
+	item_state = "powerfist"
+	flags = CONDUCT
+	attack_verb = list("whacked", "fisted", "power-punched")
+	force = 12
+	throwforce = 10
+	throw_range = 7
+	w_class = 3
+	origin_tech = "combat=5;powerstorage=3;syndicate=3"
+	var/click_delay = 1.5
+	var/fisto_setting = 1
+	var/gasperfist = 3
+	var/obj/item/weapon/tank/tank = null //Tank used for the gauntlet's piston-ram.
+
+
+/obj/item/weapon/melee/powerfist/Destroy()
+	if(tank)
+		qdel(tank)
+		tank = null
+	return ..()
+
+/obj/item/weapon/melee/powerfist/examine(mob/user)
+	..()
+	if(!in_range(user, src))
+		to_chat(user, "<span class='notice'>You'll need to get closer to see any more.</span>")
+		return
+	if(tank)
+		to_chat(user, "<span class='notice'>\icon [tank] It has \the [tank] mounted onto it.</span>")
+
+/obj/item/weapon/melee/powerfist/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/tank))
+		if(!tank)
+			var/obj/item/weapon/tank/IT = W
+			if(IT.volume <= 3)
+				to_chat(user, "<span class='warning'>\The [IT] is too small for \the [src].</span>")
+				return
+			updateTank(W, 0, user)
+	else if(istype(W, /obj/item/weapon/wrench))
+		switch(fisto_setting)
+			if(1)
+				fisto_setting = 2
+			if(2)
+				fisto_setting = 3
+			if(3)
+				fisto_setting = 1
+		playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
+		to_chat(user, "<span class='notice'>You tweak \the [src]'s piston valve to [fisto_setting].</span>")
+	else if(istype(W, /obj/item/weapon/screwdriver))
+		if(tank)
+			updateTank(tank, 1, user)
+
+
+/obj/item/weapon/melee/powerfist/proc/updateTank(obj/item/weapon/tank/emergency_oxygen/thetank, removing = 0, mob/living/carbon/human/user)
```

Comment:
you idiot, Fox, why did you typecast this to only emergency oxygen tanks when it accepts all of them.